test/e2e_node/kubelet_authz_test.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	authorizationv1 "k8s.io/api/authorization/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+	"k8s.io/kubernetes/pkg/cluster/ports"
+	"k8s.io/kubernetes/test/e2e/feature"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2eauth "k8s.io/kubernetes/test/e2e/framework/auth"
+)
+
+var _ = SIGDescribe("Kubelet Authz", feature.KubeletFineGrainedAuthz, func() {
+	f := framework.NewDefaultFramework("kubelet-authz-test")
```

Comment:
Is it going to be a problem to reuse the same framework (and namespace) for all 3 tests? I can't remember exactly when the setup is rerun, and when cleanup happens. Is there a risk that the bindings from the other cases persist across the cases?

Reply:
I create a serviceAccount and clusterRole per test and the name of the service account and clusterrole is based on the subresource. In the first test they will be called healthz, in the second case they will be called proxy and in the third case they will be called configz.

This way even if they persist they will not impact each other.